poolapi/apijson.go

```diff
@@ -0,0 +1,41 @@
+package poolapi
+
+type Response struct {
+	Status  string      `json:"status"`
+	Message string      `json:"message"`
+	Data    interface{} `json:"data,omitempty"`
```

Comment:
if a client unmashals a response, and this struct has the field `Data interface{}` the underlying concrete type will be a `map[string]interface{}`.  The json decoder has no idea what the expected data is so it doesn't know it should be unmarshaled into a `PurchaseInfo` or `Stats`.

if the field is a `*json.RawMessage` then it remains a byte slice and the client is able to do a second unmarshal to fill a `Stats` or `PurchaseInfo` or whatever else we end up adding.  This is both faster and easier than going from json -> `map[string]interface{}` -> `ExpectedType`.

Making it a `interface{}` is the easiest type to use on the server side (just set the field to the `Stats` or `PurchaseInfo` and let the json package do the rest), and does not enforce that all results are json objects, which may or may not be useful depending on what other methods are added later.

So `interface{}` is probably best to use here, on the server end.  Clients will probably want it to be `*json.RawMessage`.  Clients can either declare that type themselves or we can provide it for them along with the server's type.

The server's type can probably also be unexported, since clients won't find a whole lot of use from it.